modules/dag-l0/src/main/scala/io/constellationnetwork/dag/l0/config/types.scala

```diff
@@ -114,47 +135,63 @@ object types {
       case epoch if epoch < EpochProgress(1336392L) =>
         ProgramsDistributionConfig(
           weights = Map(
-            stardustPrimary -> Weight(5L),
-            stardustSecondary -> Weight(5L),
-            softStaking -> Weight(20L),
-            testnet -> Weight(1L),
-            dataPool -> Weight(65L)
+            stardustPrimary -> NonNegFraction.unsafeFrom(5L, 100L),
+            stardustSecondary -> NonNegFraction.unsafeFrom(5L, 100L),
+            softStaking -> NonNegFraction.unsafeFrom(20L, 100L),
+            testnet -> NonNegFraction.unsafeFrom(1L, 100L),
+            dataPool -> NonNegFraction.unsafeFrom(65L, 100L)
           ),
-          remainingWeight = Weight(4L) // facilitators
+          validatorsWeight = NonNegFraction.unsafeFrom(4L, 100L),
+          delegatorsWeight = NonNegFraction.unsafeFrom(0L, 100L)
         )
       case epoch if epoch < EpochProgress(1352274L) =>
         ProgramsDistributionConfig(
           weights = Map(
-            stardustPrimary -> Weight(5L),
-            stardustSecondary -> Weight(5L),
-            testnet -> Weight(5L),
-            dataPool -> Weight(55L)
+            stardustPrimary -> NonNegFraction.unsafeFrom(5L, 100L),
+            stardustSecondary -> NonNegFraction.unsafeFrom(5L, 100L),
+            testnet -> NonNegFraction.unsafeFrom(5L, 100L),
+            dataPool -> NonNegFraction.unsafeFrom(55L, 100L)
           ),
-          remainingWeight = Weight(30L) // facilitators
+          validatorsWeight = NonNegFraction.unsafeFrom(30L, 100L),
+          delegatorsWeight = NonNegFraction.unsafeFrom(0L, 100L)
         )
       case epoch if epoch < EpochProgress(1947530L) =>
         ProgramsDistributionConfig(
           weights = Map(
-            stardustNewPrimary -> Weight(5L),
-            stardustSecondary -> Weight(5L),
-            testnet -> Weight(3L),
-            dataPool -> Weight(55L),
-            integrationNet -> Weight(15L)
+            stardustNewPrimary -> NonNegFraction.unsafeFrom(5L, 100L),
+            stardustSecondary -> NonNegFraction.unsafeFrom(5L, 100L),
+            testnet -> NonNegFraction.unsafeFrom(3L, 100L),
+            dataPool -> NonNegFraction.unsafeFrom(55L, 100L),
+            integrationNet -> NonNegFraction.unsafeFrom(15L, 100L)
+          ),
+          validatorsWeight = NonNegFraction.unsafeFrom(17L, 100L),
+          delegatorsWeight = NonNegFraction.unsafeFrom(0L, 100L)
+        )
+      case epoch if epoch < EpochProgress(2000000L) => // TODO - change value at launch - Marks transition to new base case
```

Comment:
Need to review the code in the repo itself to double check, but is this accounting for separate values for integration vs. mainnet?